### PR TITLE
Add preview and code refacotoring and make upload image works

### PR DESCRIPTION
**Pull request by jacklam718**
#### Summary
* Add preview page (copied from @indiejoseph)
* Make image upload works
* Code refactoring
* Fix Xcode can't build issue and remove `Resources` and `Recovered References` from Xcode
* Logout by calling Facebook LoginManager.logOut method
* Update set `uid` in prediction action
* Refactor `store`

##### Note
Added new packages:
* `react-native-fetch-blob`
* `react-native-image-resizer`
* `react-native-snap-carousel`
* `redux-persist`

### CODE REVIEW

File: src/views/PreviewView.js [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+  item: {
+    prop: number,
+    chineseName2: string;
+    type: string,
+    species: string,
+    previewUrl: string,
+    title: string
+  }
+}
+const CarouselItem = ({ item }: Item, parallaxProps: Object) => (
```
Comment (indiejoseph) [inline src/views/PreviewView.js]: Can move to separate file?

Comment (jacklam718) [reply src/views/PreviewView.js]: move to where? any advice on this?

Comment (indiejoseph) [reply src/views/PreviewView.js]: As a static component

File: src/common/imageUtil.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      offset,
+      size,
+    },
+    _uri => resolve(_uri),
+    err => reject(err));
+  })
+);
+
+export const resizeImage = (uri, { size, format, quality } = {
+  size: parseInt(Config.IMAGE_SIZE, 10) || 960,
```
Comment (indiejoseph) [inline src/common/imageUtil.js]: Please use 1440 instead of 960

**Review by indiejoseph — APPROVED** [review]

**Pull request merged**